src/node/routes/login.ts

```diff
@@ -12,16 +12,20 @@ export enum Cookie {
 }
 
 // RateLimiter wraps around the limiter library for logins.
-// It allows 2 logins every minute and 12 logins every hour.
-class RateLimiter {
+// It allows 2 logins every minute plus 12 logins every hour.
+export class RateLimiter {
   private readonly minuteLimiter = new Limiter(2, "minute")
   private readonly hourLimiter = new Limiter(12, "hour")
 
+  public canTry(): boolean {
+    return this.minuteLimiter.getTokensRemaining() > 0 || this.hourLimiter.getTokensRemaining() > 0
+  }
+
   public try(): boolean {
-    if (this.minuteLimiter.tryRemoveTokens(1)) {
-      return true
+    if (this.canTry()) {
```

Comment:
Nit: we don't need the `canTry` here I think? Should be the same behavior without it.

Reply:
Oh hmm, yeah - in this case this is true; with `&&` you might end up with removing a token from one limiter but not the other, but with `||` there's no need for it.

Reply:
nice catch! i'll fix